### ISSUE TITLE
No response from rasa x server ....Pika.exceptions.ConnectionClosedByBroker: 

### ISSUE THREAD

**Issue by gonhonoria** [RasaHQ/rasa#5878]
**Rasa version**:
RASA_X_VERSION=0.28.3
RASA_VERSION=1.10.0
RASA_X_DEMO_VERSION=0.28.1
**Python version**: 3.6

**Operating system** (windows, osx, ...): Ubuntu 18.04 LTS

**Issue**:
I was testing my assistant on rasa X , but i got not feedback from the server action. While checking the logs i have seen below 

**Error (including full traceback)**:
```      | ERROR:pika.adapters.blocking_connection:Unexpected connection close detected: ConnectionClosed
ByBroker: (320) "CONNECTION_FORCED - broker forced connection closure with reason 'shutdown'"
rasa-x_1           | ERROR:rasax.community.services.event_service:Caught an exception while consuming events. Will 
retry in 5 s.
rasa-x_1           | Traceback (most recent call last):
rasa-x_1           |   File "/usr/local/lib/python3.7/site-packages/rasax/community/services/event_service.py", lin
e 1631, in continuously_consume
rasa-x_1           |     consumer.consume()
rasa-x_1           |   File "/usr/local/lib/python3.7/site-packages/rasax/community/services/event_consumers/pika_c
onsumer.py", line 180, in consume
rasa-x_1           |     self.channel.start_consuming()
rasa-x_1           |   File "/usr/local/lib/python3.7/site-packages/pika/adapters/blocking_connection.py", line 186
6, in start_consuming
rasa-x_1           |     self._process_data_events(time_limit=None)
rasa-x_1           |   File "/usr/local/lib/python3.7/site-packages/pika/adapters/blocking_connection.py", line 202
7, in _process_data_events
rasa-x_1           |     self.connection.process_data_events(time_limit=time_limit)
rasa-x_1           |   File "/usr/local/lib/python3.7/site-packages/pika/adapters/blocking_connection.py", line 825
, in process_data_events
rasa-x_1           |     self._flush_output(common_terminator)
rasa-x_1           |   File "/usr/local/lib/python3.7/site-packages/pika/adapters/blocking_connection.py", line 522
, in _flush_output
rasa-x_1           |     raise self._closed_result.value.error
rasa-x_1           | pika.exceptions.ConnectionClosedByBroker: (320, "CONNECTION_FORCED - broker forced connection 
closure with reason 'shutdown'")
rasa-x_1           | INFO:rasax.community.services.event_consumers.pika_consumer:Start consuming queue 'rasa_produc
tion_events' on pika host 'rabbit'.

```

**Command or request that led to error**:
```

```

**Content of configuration file (config.yml)** (if relevant):
```yml
language: en
pipeline:
  - name: SpacyNLP
    model: en_core_web_md
    case_sensitive: false
  - name: SpacyTokenizer
  - name: SpacyFeaturizer
  - name: RegexFeaturizer
  - name: CRFEntityExtractor
  - name: EntitySynonymMapper
  - name: SklearnIntentClassifier
policies:
  - name: MemoizationPolicy
  - name: TEDPolicy
    max_history: 5
    epochs: 150
  - name: MappingPolicy
  - name: TwoStageFallbackPolicy

```

**Content of domain file (domain.yml)** (if relevant):
```yml
session_config:
  session_expiration_time: 60
  carry_over_slots_to_new_session: true
intents:
  - greet
  - goodbye
  - affirm
  - deny
  - mood_great
  - mood_unhappy
  - bot_challenge
  - inform
  - user_profile
  - thanks
  - out_of_scope
  - rate
  - affirm_search_job_general
  - affirm_search_job_questions
entities:
  - job_title
  - name
  - age
  - mail
slots:
  age:
    type: unfeaturized
  job_title:
    type: unfeaturized
  location:
    type: unfeaturized
  mail:
    type: unfeaturized
  name:
    type: unfeaturized
  verdict:
    type: categorical
    values:
      - continue
      - ok
      - almost
      - not ok
responses:
  utter_greet:
    - text: Hey!
    - text: Hello!
  utter_did_that_help:
    - text: Did that help you?
  utter_happy:
    - text: 'Great, carry on!'
  utter_goodbye:
    - text: Bye
  utter_iamabot:
    - text: 'I am a bot, powered by Rasa.'
  utter_botdescription:
    - text: >-
        I am jobfinder for mybusiness company. If you are interested in joining
        us, I can help you find the job offer that best suits you according to
        your preferences and skills. Would you like to start?
    - text: >-
        I am jobfinder. I will help you find the most suitable job offer from
        available positions in mybusiness company. Are you ready?
  utter_userdescription:
    - text: 'Kindly,tell me a little about yourself'
    - text: 'Please, briefly describe yourself'
  utter_background:
    - text: Walk Me Through Your Resume.
    - text: Tell me about your professional background.
  utter_expertize:
    - text: What skills have you gained from your work experience and education?
    - text: What are your expertise and qualifications?
  utter_ask_location:
    - text: Do you have any location preference?
    - text: Are you interested in a specific location?
    - text: Do You Have a Geographic Preference?
  utter_ask_name:
    - text: Let's get started! What's your name?
    - text: Great! What's your name?
    - text: 'Please,give me your name!'
  utter_ask_mail:
    - text: What is your e-mail address?
    - text: 'Please, provide me your e-mail address!'
  utter_ask_rephrase:
    - text: I am so sorry I do not understand your point. Could you say it again?
    - text: I didn't get that. Could you rephrase please?
    - text: I am so sorry I do not understand your point.Please be more specific!
    - text: Can you say that a different way?
    - text: >-
        I am jobfinder. I can help you find the most suitable job offer from
        available positions in mybusiness company. Would you like to start?
actions:
  - utter_greet
  - utter_did_that_help
  - utter_happy
  - utter_goodbye
  - utter_iamabot
  - utter_botdescription
  - utter_userdescription
  - utter_background
  - utter_expertize
  - utter_ask_location
  - utter_ask_name
  - utter_ask_mail
  - action_interview
  - action_bonus
  - action_register_name
  - action_register_mail
  - utter_ask_rephrase
  - action_job_details
```

You can find more details at
https://github.com/gonhonoria/HireBotProject
![image](https://user-images.githubusercontent.com/37090872/82678861-e064e900-9c39-11ea-8bce-3acdde6c4906.png)

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @tmbo will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by gonhonoria** [comment]
no feedback. why?


Le lun. 25 mai 2020 à 06:39, Sara-tagger <notifications@github.com> a
écrit :

**Comment by pavlinakoutecka** [comment]
Hi,

I am facing a similar problem. I am trying to run both Rasa X and custom actions server on an Ubuntu remote server. While running the `rasa run actions` with simple `rasa shell` command is working perfectly fine, using the Rasa X server quick-install script with built action server image is not working. I suppose then, the error is not in my custom action `actions.py` script.

First, I created a custom action server image following this tutorial: https://rasa.com/docs/rasa/user-guide/how-to-deploy/#building-an-action-server-image. After that, I configured the Rasa X using the server quick-install following this tutorial: https://rasa.com/docs/rasa-x/installation-and-setup/customize/. I checked the Kubernetes logs, it seems that Rasa successfully pulled the right custom action server image.

After accessing the Rasa X, I uploaded a trained model. Everything works as it should, but running into some of my custom actions, there is no response. 

![image](https://user-images.githubusercontent.com/64008331/89182324-d0507a80-d595-11ea-9df6-0a1a2342f27a.png)

As you can see, the `None` fields should be filled with an answer obtained from some custom action. 

![image](https://user-images.githubusercontent.com/64008331/89182527-29b8a980-d596-11ea-944e-9e980b91f77f.png)

This is the output using the `rasa shell` command for the exact same custom actions, model, and Rasa configuration.

Thanks in advance for your answer :blush: .

**Comment by robynbitner** [comment]
Any updates?

**Comment by mprazz** [comment]
@edmacovaz Can we get a bit of context on why this issue was moved to the Infra board? This looks related to pika. Seems to be for @RasaHQ/scale-squad or @RasaHQ/enable-squad 🤔

**Comment by edmacovaz** [comment]
@mprazz Sounds like a mistake on my part (all I know about Pikas is that they're cute and endangered). I'll move it to Enable.

**Comment by wochinge** [comment]
I'll close this issue as we have replaced the pika library since then and did massive changes in general to the module.
